### PR TITLE
Enable clang-format AlignConsecutiveMacros.

### PR DESCRIPTION
**Pull request by StephanTLavavej**
# Description

* Update `.clang-format` defaults.

* Enable `AlignConsecutiveMacros`, newly supported by clang-format 9. We often followed this style before using clang-format, and I believe that it produces generally positive changes.

* Apply clang-format changes.

* Replace `_FITS` and `_NEXT_ALIGN` macros. clang-format didn't damage them, it just brought them to my attention. These macros weren't improving clarity, and the code is more systematic without them. (For the time being, I have refrained from further overhauling the metaprogramming here.)

# Checklist

Be sure you've read README.md and understand the scope of this repo.

If you're unsure about a box, leave it unchecked. A maintainer will help you.

- [x] Identifiers in product code changes are properly `_Ugly` as per
  https://eel.is/c++draft/lex.name#3.1 or there are no product code changes.
- [ ] The STL builds successfully and all tests have passed (must be manually
  verified by an STL maintainer before automated testing is enabled on GitHub,
  leave this unchecked for initial submission).
- [x] These changes introduce no known ABI breaks (adding members, renaming
  members, adding virtual functions, changing whether a type is an aggregate
  or trivially copyable, etc.).
- [x] These changes were written from scratch using only this repository and
  the C++ Working Draft as a reference (and any other cited standards).
  If they were derived from a project that's already listed in NOTICE.txt,
  that's fine, but please mention it. If they were derived from any other
  project (including Boost and libc++, which are not yet listed in
  NOTICE.txt), you *must* mention it here, so we can determine whether the
  license is compatible and what else needs to be done.

### CODE REVIEW

File: stl/inc/type_traits [hunk 1]
```diff
@@ -1098,27 +1095,32 @@ struct _Aligned<_Len, _Align, double, false> {
 
 template <size_t _Len, size_t _Align>
 struct _Aligned<_Len, _Align, int, false> {
-    _NEXT_ALIGN(double);
```
Comment (barcharcraz) [inline stl/inc/type_traits]: this looks like a perhaps unrelated change? how'd this get here?

File: stl/inc/type_traits [hunk 2]
```diff
@@ -1098,27 +1095,32 @@ struct _Aligned<_Len, _Align, double, false> {
...
 
 template <size_t _Len, size_t _Align = alignof(max_align_t)>
 struct aligned_storage { // define type with size _Len and alignment _Align
-    using type = typename _Aligned<_Len, _Align, char, _FITS(char)>::type;
+    using _Next                 = char;
+    static constexpr bool _Fits = _Align <= alignof(_Next);
+    using type                  = typename _Aligned<_Len, _Align, _Next, _Fits>::type;
 };
 
-#undef _FITS
```
Comment (barcharcraz) [inline stl/inc/type_traits]: ditto on how did this get here.

File: stl/src/xtime.cpp [hunk 3]
```diff
@@ -8,10 +8,10 @@
 #include <time.h>
 #include <xtimec.h>
 
-#define NSEC_PER_SEC 1000000000L
+#define NSEC_PER_SEC  1000000000L
 #define NSEC_PER_MSEC 1000000L
 #define NSEC_PER_USEC 1000L
-#define MSEC_PER_SEC 1000
```
Comment (barcharcraz) [inline stl/src/xtime.cpp]: not a change for this PR, but we could make these const or constexpr instead of macros (or just write them out, it's not like this will ever change).

File: stl/inc/type_traits (hunk 1 above)
Comment (StephanTLavavej) [reply stl/inc/type_traits]: Please see the description - it is indeed an unrelated cleanup. Would you like me to separate it, which I can do trivially?

File: stl/src/xtime.cpp (hunk 3 above)
Comment (StephanTLavavej) [reply stl/src/xtime.cpp]: I will file an enhancement issue.

File: stl/inc/type_traits (hunk 1 above)
Comment (barcharcraz) [reply stl/inc/type_traits]: nah, just something to keep in mind. Sorry, I skipped right to the code since you described it in standup :D.

File: stl/src/xtime.cpp (hunk 3 above)
Comment (StephanTLavavej) [reply stl/src/xtime.cpp]: Filed as #270.
